bvm/ballerina-core/src/main/java/org/ballerinalang/bre/bvm/CPU.java

```diff
@@ -3446,16 +3446,15 @@ private static void calculateLength(WorkerExecutionContext ctx, int[] operands,
         if (typeTag == TypeTags.XML_TAG) {
             sf.longRegs[j] = ((BXML) entity).length();
             return;
-        } else if (typeTag == TypeTags.MAP_TAG) {
+        } else if (entity instanceof BMap) {
             sf.longRegs[j] = ((BMap) entity).size();
             return;
-        } else if (!(entity instanceof BNewArray)) {
-            sf.longRegs[j] = -1;
+        } else if (entity instanceof BNewArray) {
+            sf.longRegs[j] = ((BNewArray) entity).size();
             return;
         }
 
-        BNewArray newArray = (BNewArray) entity;
-        sf.longRegs[j] = newArray.size();
+        sf.longRegs[j] = -1;
```

Comment:
I am not sure giving -1 is correct for single item json values. ( json j = "foo"; ). Given that lengthof operation is going to be removed and new length function is not supported for json type, we may not need to worry about this. But please create an issue, so we won't miss this. @SupunS  pls note.

Reply:
Created https://github.com/ballerina-platform/ballerina-lang/issues/10239 to track.